js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Node.tsx

```diff
@@ -370,3 +391,17 @@ function StatusDot({node}: {node: Pick<GraphNode, 'assetKey' | 'definition'>}) {
   });
   return <StatusCaseDot statusCase={status.case} />;
 }
+
+function isElementInsideSVGViewport(element: Element | null) {
+  if (!element) {
+    // We've reached the root without finding an <svg> element
+    return false;
+  }
+
+  if (element.classList.contains('svgViewport')) {
+    return true;
+  }
+
+  // Start the recursive check
+  return isElementInsideSVGViewport(element.parentElement);
```

Comment:
I think that this is a bit easier than this actually! You can use `element.closest('.svgViewport')` to do this recurisve scan up the tree + return the closest parent (or `[data-svg-viewport]` if you apply that other comment :)

Reply:
Ah yeah I was thinking of using closest, originally I was doing something else (I was looking ownerSVGElement) but then realized that the focused element was outside of the svg viewport for some reason (guess we don't actually focus the foreignNode inside of the SVGViewport component)

Reply:
^ which we totally should do to make the node itself keyboard navigable